features/activity/cancel_try_cancel/feature.java

```diff
@@ -0,0 +1,116 @@
+package activity.cancel_try_cancel;
+
+import io.temporal.activity.Activity;
+import io.temporal.activity.ActivityCancellationType;
+import io.temporal.activity.ActivityInterface;
+import io.temporal.activity.ActivityMethod;
+import io.temporal.client.ActivityCanceledException;
+import io.temporal.client.WorkflowClient;
+import io.temporal.common.RetryOptions;
+import io.temporal.failure.ActivityFailure;
+import io.temporal.failure.ApplicationFailure;
+import io.temporal.failure.CanceledFailure;
+import io.temporal.sdkfeatures.Feature;
+import io.temporal.sdkfeatures.Run;
+import io.temporal.sdkfeatures.Runner;
+import io.temporal.sdkfeatures.SimpleWorkflow;
+import io.temporal.workflow.*;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicReference;
+
+@ActivityInterface
+public interface feature extends Feature, SimpleWorkflow {
+  @SignalMethod
+  void activityResult(String result);
+
+  @ActivityMethod
+  void cancellableActivity();
+
+  class Impl implements feature {
+    private CompletablePromise<String> activityResult;
+
+    @Override
+    public void workflow() {
+      activityResult = Workflow.newPromise();
+      var activities = activities(feature.class, builder -> builder
+              .setScheduleToCloseTimeout(Duration.ofMinutes(1))
+              .setHeartbeatTimeout(Duration.ofSeconds(5))
+              // Disable retry
+              .setRetryOptions(RetryOptions.newBuilder().setMaximumAttempts(1).build())
```

Comment:
Does that actually matter here? Shouldn't cancel error be non-retryable anyway? (Noticed this applies to all of them)

Reply:
Hmm.. yeah, this would make the test flakier.
Not a concern now but eventually it might.

Reply:
In my head "if I don't want an activity to ever retry, I don't let it". Basically I want any failure of the activity for these tests to be considered non-retryable, not just the expected one.